src/share/classes/com/sun/javatest/regtest/config/RegressionTestSuite.java

```diff
@@ -215,10 +215,26 @@ public boolean useOtherVM(TestDescription td) throws TestSuite.Fault {
         return properties.useOtherVM(td.getFile());
     }
 
+    /**
+     * @param td the test description
+     * {@return true if the test is configured to run exclusively, false otherwise}
+     * @deprecated Use {@link #isExclusiveAccessRequired(TestDescription)} instead,
+     *             which doesn't throw the checked exception
+     * @throws TestSuite.Fault
+     */
+    @Deprecated
     public boolean needsExclusiveAccess(TestDescription td) throws TestSuite.Fault {
```

Comment:
This already existing `throws TestSuite.Fault` is actually unnecessary because nothing in that logic throws that checked exception. However, I decided not to change this method signature since that can cause compatibility issue - I'm guessing this `com.sun.javatest.regtest.config.RegressionTestSuite` is a public API?
That's why I decided to introduce a new method which doesn't throw the checked exception.

Reply:
I am not aware of any consumer of `RegressionTestSuite` - is there, @jonathan-gibbons ?

Reply:
`RegressionTestSuite` is a mostly-private implementation of a public interface `TestSuite`.
"mostly private" means that there are probably no other uses of the class. Maybe at one point I thought there might be.

I also suspect this might be a historical leftover from times past, when access to the underlying properties might have been more lazy.

At any rate, any clients of `RegressionTestSuite` would be "friends" and should be willing to tolerate change. There is no published commitment to the stability of these internal classes.

Reply:
> At any rate, any clients of RegressionTestSuite would be "friends" and should be willing to tolerate change. There is no published commitment to the stability of these internal classes.

That makes things simpler and cleaner too. I've now updated the PR to not introduce another method and instead just update the signature of the existing one.